lib/logmsg/logmsg-serialize.c

```diff
@@ -45,6 +45,15 @@ _setup_ts_processed(UnixTime *timestamps, const UnixTime *processed)
     timestamps[LM_TS_PROCESSED] = timestamps[LM_TS_RECVD];
 }
 
+static void
+nv_table_serialize_with_compaction(LogMessageSerializationState *state, NVTable *old)
+{
+  NVTable *payload;
+  payload = nv_table_compact(old);
+  nv_table_serialize(state, payload);
+  nv_table_unref(payload);
```

Comment:
This on the other hand means possible leaking the payload when compact is turned on.
e.g.: two thread sees `ref_cnt=2`, does unref and `ref_cnt` became 1 so no free is "needed".

Despite this I would merge it as is, as fixes a crash compared to a leak (that is present in the prev release anyway, and the feature triggering it is turned off by default).

Reply:
Oh sorry my bad, payload here is the new `NVTable`, local variable so no issue here.

Reply:
maybe we should rename `nv_table_compact` to `nv_table_clone_compact` ...